src/app/clusters/webrtc-transport-provider-server/tests/TestWebRTCTransportProviderCluster.cpp

```diff
@@ -0,0 +1,626 @@
+    EXPECT_EQ(outSession.streamUsage, StreamUsageEnum::kLiveView);
+    EXPECT_FALSE(outDeferredOffer);
+
+    // Test deferred offer
+    mockDelegate.SetDeferredOffer(true);
+    result = mockDelegate.HandleSolicitOffer(args, outSession, outDeferredOffer);
+    EXPECT_EQ(result, CHIP_NO_ERROR);
+    EXPECT_TRUE(outDeferredOffer);
+
+    // Test error case
+    mockDelegate.SetSolicitOfferResult(CHIP_ERROR_NO_MEMORY);
+    result = mockDelegate.HandleSolicitOffer(args, outSession, outDeferredOffer);
+    EXPECT_EQ(result, CHIP_ERROR_NO_MEMORY);
+}
+
+TEST_F(TestWebRTCTransportProviderCluster, TestDelegateHandleProvideOffer)
+{
+    MockWebRTCTransportProviderDelegate mockDelegate;
+    WebRTCTransportProviderCluster server(kTestEndpointId, mockDelegate);
+
+    // Setup test data
+    Delegate::ProvideOfferRequestArgs args;
+    args.sessionId   = 456;
+    args.streamUsage = StreamUsageEnum::kRecording;
+    args.peerNodeId  = 0x5678ULL;
+    args.fabricIndex = 1;
+    args.sdp         = "test_sdp_offer";
+
+    WebRTCSessionStruct outSession;
+
+    // Test successful provide offer
+    mockDelegate.SetProvideOfferResult(CHIP_NO_ERROR);
+
+    CHIP_ERROR result = mockDelegate.HandleProvideOffer(args, outSession);
+    EXPECT_EQ(result, CHIP_NO_ERROR);
+    EXPECT_EQ(outSession.id, 456);
+    EXPECT_EQ(outSession.streamUsage, StreamUsageEnum::kRecording);
+    EXPECT_EQ(mockDelegate.GetLastProvideOfferArgs().sdp, "test_sdp_offer");
+
+    // Test error case
+    mockDelegate.SetProvideOfferResult(CHIP_ERROR_INVALID_ARGUMENT);
+    result = mockDelegate.HandleProvideOffer(args, outSession);
+    EXPECT_EQ(result, CHIP_ERROR_INVALID_ARGUMENT);
+}
+
+TEST_F(TestWebRTCTransportProviderCluster, TestDelegateHandleProvideAnswer)
+{
+    MockWebRTCTransportProviderDelegate mockDelegate;
+    WebRTCTransportProviderCluster server(kTestEndpointId, mockDelegate);
+
+    // Setup test data
+    uint16_t sessionId        = 789;
+    std::string testSdpAnswer = "test_sdp_answer";
+
+    // Test successful answer handling
+    mockDelegate.SetProvideAnswerResult(CHIP_NO_ERROR);
+
+    CHIP_ERROR result = mockDelegate.HandleProvideAnswer(sessionId, testSdpAnswer);
+    EXPECT_EQ(result, CHIP_NO_ERROR);
+    EXPECT_EQ(mockDelegate.GetLastSessionId(), 789);
+    EXPECT_EQ(mockDelegate.GetLastSdpAnswer(), testSdpAnswer);
+
+    // Test error case
+    mockDelegate.SetProvideAnswerResult(CHIP_ERROR_INTERNAL);
+    result = mockDelegate.HandleProvideAnswer(sessionId, testSdpAnswer);
+    EXPECT_EQ(result, CHIP_ERROR_INTERNAL);
+}
+
+TEST_F(TestWebRTCTransportProviderCluster, TestDelegateHandleProvideICECandidates)
+{
+    MockWebRTCTransportProviderDelegate mockDelegate;
+    WebRTCTransportProviderCluster server(kTestEndpointId, mockDelegate);
+
+    // Setup test data
+    uint16_t sessionId = 999;
+    std::vector<ICECandidateStruct> testCandidates;
+
+    ICECandidateStruct candidate1;
+    candidate1.candidate = chip::CharSpan("candidate1", 10);
```

Comment:
```suggestion
    candidate1.candidate = "candidate1"_span;
```

is probably prefered throghout